crates/uv/src/commands/tool/install.rs

```diff
@@ -76,24 +79,28 @@ pub(crate) async fn install(
         .connectivity(connectivity)
         .native_tls(native_tls);
 
-    // Resolve the `from` requirement.
-    let from = if let Some(from) = from {
-        // Parse the positional name. If the user provided more than a package name, it's an error
-        // (e.g., `uv install foo==1.0 --from foo`).
-        let Ok(package) = PackageName::from_str(&package) else {
-            bail!("Package requirement (`{from}`) provided with `--from` conflicts with install request (`{package}`)", from = from.cyan(), package = package.cyan())
-        };
+    // Parse the input requirement.
+    let target = Target::parse(&package, from.as_deref());
 
-        let source = if editable {
-            RequirementsSource::Editable(from)
-        } else {
-            RequirementsSource::Package(from)
-        };
-        let requirements = RequirementsSpecification::from_source(&source, &client_builder)
-            .await?
-            .requirements;
+    // If the user passed, e.g., `ruff@latest`, refresh the cache.
+    let cache = if target.is_latest() {
+        cache.with_refresh(Refresh::All(Timestamp::now()))
```

Comment:
Refresh all? Or just refresh Ruff?

Reply:
In `uv tool run` at least we refresh everything. I think we might have to refresh everything here because we don't know the package name yet...?

Reply:
Okay. Seems fine.